### ISSUE TITLE
interval diff.in_years() is returning incorrect values

### ISSUE THREAD

**Issue by xBlackBartx** [sdispater/pendulum#70]
using the following code: ** Note this was tested on 11/5/2016, so result should be 15, instead it is returning 16

    from pendulum import Pendulum as p
    d1 = p(2000,11,20) 
    diff = d1.diff()
    diff.in_years()
    16

Trial and error shows that it reported 15 when I set the date to 2000,11,25 meaning that it was 10 days off.

**Comment by AndreasBackx** [comment]
@xBlackBartx this is because Pendulum rounds `Interval.total_years` (and other `Interval.total_` methods). You can implement your own by rounding it down to have the return value you desire. See the implementation of `Interval.total_years`. I would say the rounded behaviour that is being used right now is the most desirable.

**Comment by sdispater** [comment]
@xBlackBartx @AndreasBackx `pendulum` will actually truncate the `total_years()` value in `in_years()`. `total_years()` is just an approximation since getting a `timedelta` in years cannot always be accurate due to leap years.

```python
>>> import pendulum

>>> d = pendulum.create(2000, 11, 20)
>>> diff = d.diff()
>>> diff.in_years()
16
>>> diff.total_years()
16.0

>>> d = pendulum.create(2000, 11, 25)
>>> diff = d.diff()
>>> diff.in_years()
15
>>> diff.total_years()
15.9
```

In the documentation, in the [Difference](https://pendulum.eustace.io/docs/#difference) section, the behavior is described like this:

> The `diff()` method returns a [Period](https://pendulum.eustace.io/docs/#period) instance that represents the total duration between two `Pendulum` instances. This interval can be then expressed in various units. These interval methods always return the total difference expressed in the specified time requested. **All values are truncated and not rounded**.

What this means is that every `in_*()` methods will give you a truncated value. So, if you use the `in_months()` and `in_years()` don't expect an *accurate* value since it's just not possible.

**Comment by danilobellini** [comment]
The `in_years` method already truncates (it doesn't round), since its code is just `int(self.total_days())` as of today. But `total_month` and `total_year` rounds to 1 decimal digit. BTW, why the `total_month` and `total_year` methods are rounding the result?

```
In [1]: import pendulum

In [2]: dt1 = pendulum.create(2000, 11, 20, tz="America/New_York")

In [3]: dt2 = pendulum.create(2000, 11, 25, tz="America/New_York")

In [4]: dt3 = pendulum.create(2016, 11, 5, tz="America/New_York")

In [5]: (dt3 - dt1).total_years()
Out[5]: 16.0

In [6]: (dt3 - dt2).total_years()
Out[6]: 15.9
```

There's also the `(dt3 - dt2).years`, but it's 16 in both cases, not 15. Shouldn't all these year intervals be 15? I mean, from `"2000-11-06"` to `"2016-11-05"` there are still only 15 full years, not 16, I'd be willing to do this to get one from another:

```
pendulum.create(2000, 11, 6).add(years=15, months=11, days=30)
```

This triple `years=15, months=11, days=30` seem to be the diff one is expecting when talking about a date diff, as those dates still didn't complete the same day in the new year.

The `total_*` without the rounding methods are fine when they gives the time interval for some rough conversion to days/weeks/months/years. But I'd argue it's not something always expected when dealing with dates, and that's probably what OP wanted with the `in_*` methods: Not all days have 24 hours, not all years have 365 days and the month duration in days are far from constant, but when dealing with date intervals one probably doesn't want to mix these concepts.

Also, shouldn't `dt2 + (dt3 - dt2)` and `dt1 + (dt3 - dt1)` be equal to `dt3`?

**Comment by sdispater** [comment]
@danilobellini You have a point.

The thing is when I started implemented it the `Interval` class, like the native `timedelta` class, would only store `days`, `seconds` and `microseconds`. I introduced `minutes`, `hours`, `weeks`, `months` and `years` as read units to ease things. However, since intervals are not datetime aware I had to provide approximations for years and months.

All this was before I implemented the `Period` class that is datetime aware. So it's theoretically possible to provide accurate units since we know the starting and ending point of the interval in this case.

I will take a look at it to see if I can find a way to make it work.

**Comment by sdispater** [comment]
I pushed a fix to the `develop` branch (8be054c17a1c4164b04b0112c7ad9226824dc060) that addresses this issue. You can test it if you want but be aware that a lot of changes have been made on the `develop` branch so things might break, even though it should be limited to intervals.

Anyway, with the fix you get:

```python
>>> import pendulum

>>> dt1 = pendulum.create(2000, 11, 20, tz="America/New_York")
>>> dt2 = pendulum.create(2000, 11, 25, tz="America/New_York")
>>> dt3 = pendulum.create(2016, 11, 5, tz="America/New_York")

>>> (dt3 - dt1).years
15
>>> (dt3 - dt1).months
11

>>> (dt3 - dt2).years
15
>>> (dt3 - dt2).months
11
```

Note that the `total_months()` and `total_years()` are now deprecated since they were bad approximations and misleading.
